### ISSUE TITLE
Question: Listeners seems to be triggered on component mount and not just when something happens

### ISSUE THREAD

**Issue by MangelMaxime** [fsprojects/Avalonia.FuncUI#435]
Hello,

I am working a "Form GUI" which does some input validation.

While doing so, I discovered that if you listen to an event then it is triggered when the component is mounted for the first time:

For example, if I define:

```fs
TextBox.create [
    TextBox.onTextChanged (fun _ ->
        printfn "text changed"
    )
]
```

Then when the application load, I will see the `"text changed"` before having pressed a key.

If I understood FuncUI code correctly, I believe all subscription/listener are impacted because they use the same code under the hood (from `AttrBuilder`).

This is an issue for me because, when loading the UI the user will see:

![CleanShot 2024-09-05 at 15 41 22](https://github.com/user-attachments/assets/70ea7708-c28a-42ff-8c1a-b642f76cd121)

And I would like the error to be displayed only if the user wrote something, otherwise all the GUI will be red at loading 😅

I saw the `SubPatchOptions` but it doesn't seems to have any effect in my use case, am I missing something?

**Comment by JaggerJo** [comment]
Hey @MangelMaxime 👋

does your code actually look just like this?

```fsharp
TextBox.create [
    TextBox.onTextChanged (fun _ ->
        printfn "text changed"
    )
]
```

or more like this?

```fsharp
TextBox.create [
    TextBox.onTextChanged (fun _ ->
        printfn "text changed"
    )
    TextBox.text "..."
]
```

Can you send me the actual code / a more complete example?

**Comment by MangelMaxime** [comment]
My code looks more like the second example:

```fs
TextBox.create [
    TextBox.text shortDescription
    TextBox.horizontalAlignment HorizontalAlignment.Stretch
    TextBox.classes [
        "short-description"
    ]
    TextBox.onTextChanged (ChangeShortDescription >> dispatch)
]
```

Full reproduction code:

Except the fake `CommitParserConfig` the code is almost a one-to-one equivalent. Note, that I am using `Avalonia.FuncUI.Elmish` to manage my state.

```fs
namespace CommitLinter.Commands.Interactive

open System
open Avalonia
open Avalonia.Controls.ApplicationLifetimes
open Avalonia.Themes.Fluent
open Avalonia.FuncUI.Hosts
open Avalonia.Controls
open Elmish
open Avalonia.FuncUI
open Avalonia.FuncUI.Elmish
open Avalonia.FuncUI.DSL
open Avalonia.Layout
open Avalonia.Media

module Elmish =

    type FormValues =
        {
            CommitTypeIndex: int
            ShortMessage: string
            Description: string
            Tags: string list
            IsBreakingChange: bool
        }

    type FormErrors =
        {
            ShortMessage: string option
            Tags: string option
        }

        member this.HasErrors = this.ShortMessage.IsSome || this.Tags.IsSome

    type Model =
        {
            FormValues: FormValues
            FormErrors: FormErrors
        }

    type Msg =
        | ChangeCommitTypeIndex of int
        | ChangeShortDescription of string
        | WriteCommit

    let init () =
        {
            FormValues =
                {
                    CommitTypeIndex = 0
                    ShortMessage = ""
                    Description = ""
                    Tags = []
                    IsBreakingChange = false
                }
            FormErrors =
                {
                    ShortMessage = None
                    Tags = None
                }
        },
        Cmd.none

    let private validateForm (model: Model) =
        let errors =
            {
                ShortMessage =
                    if String.IsNullOrWhiteSpace model.FormValues.ShortMessage then
                        Some "This field is required"
                    else
                        None
                Tags = None
            }

        { model with
            FormErrors = errors
        }

    let update msg model =
        match msg with
        | ChangeCommitTypeIndex index ->
            { model with
                FormValues =
                    { model.FormValues with
                        CommitTypeIndex = index
                    }
            },
            Cmd.none

        | ChangeShortDescription shortDescription ->
            printfn "Short description: %s" shortDescription
            { model with
                FormValues =
                    { model.FormValues with
                        ShortMessage = shortDescription
                    }
            }
            |> validateForm,
            Cmd.none

        | WriteCommit ->
            let validatedModel = validateForm model

            if validatedModel.FormErrors.HasErrors then
                validatedModel, Cmd.none
            else
                validatedModel, Cmd.none

    let fieldLabel text =
        TextBlock.create [
            TextBlock.text text
        ]

    let errorMessage text =
        TextBlock.create [
            TextBlock.text text
            TextBlock.foreground Brushes.Red
        ]

    let withLabelAndError (label: string) (errorOpt: string option) (content: Types.IView) : Types.IView =
        StackPanel.create [
            StackPanel.orientation Orientation.Vertical
            StackPanel.spacing 5
            StackPanel.children [
                fieldLabel label
                content
                match errorOpt with
                | Some text -> errorMessage text
                | None -> TextBlock.create [] // Preserve layout
            ]
        ]

    let commitTypeField (model: Model) (dispatch: Dispatch<Msg>) =
        ComboBox.create [

            // model.Config.Types |> List.map _.Name |> ComboBox.dataItems
            ComboBox.dataItems [
                "feat"
                "fix"
                "docs"
                "style"
            ]

            ComboBox.horizontalAlignment HorizontalAlignment.Stretch
            ComboBox.classes [
                "commit-type"
            ]
            ComboBox.selectedIndex model.FormValues.CommitTypeIndex
            ComboBox.onSelectedIndexChanged (ChangeCommitTypeIndex >> dispatch)
        ]
        |> withLabelAndError "Commit type" None

    let shortDescriptionField
        (shortDescription: string)
        (errorOpt: string option)
        (dispatch: Dispatch<Msg>)
        =
        TextBox.create [
            TextBox.text shortDescription
            TextBox.horizontalAlignment HorizontalAlignment.Stretch
            TextBox.classes [
                "short-description"
            ]
            TextBox.onTextChanged (ChangeShortDescription >> dispatch)
        ]
        |> withLabelAndError "Short description" errorOpt

    let view (model: Model) (dispatch: Dispatch<Msg>) =
        Grid.create [
            Grid.margin (Thickness 20)
            Grid.columnDefinitions "200, *"
            Grid.children [
                Panel.create [
                    Grid.column 0
                    Panel.children [
                        commitTypeField model dispatch
                    ]
                ]

                Panel.create [
                    Grid.column 1
                    Panel.children [
                        shortDescriptionField
                            model.FormValues.ShortMessage
                            model.FormErrors.ShortMessage
                            dispatch
                    ]
                ]

                Button.create [
                    Button.content (
                        TextBlock.create [
                            TextBlock.text "Write commit"
                        ]
                    )
                    Button.horizontalAlignment HorizontalAlignment.Right
                    Button.onClick (fun _ -> WriteCommit |> dispatch)
                ]
            ]
        ]

type CommitParserConfig = obj // Fake type for reproduction

type MainWindow(config: CommitParserConfig) as this =
    inherit HostWindow()

    do
        base.Title <- "Commit editor"
        // base.Content <- Main.view config
        base.Width <- 1024.0
        base.Height <- 768.0
        base.CanResize <- false

#if DEBUG
        this.AttachDevTools()
#endif

        Program.mkProgram Elmish.init Elmish.update Elmish.view
        |> Program.withHost this
        |> Program.withConsoleTrace
        |> Program.runWith ()
        // |> Program.runWith config // Real code would pass config here

type App(config: CommitParserConfig) =
    inherit Application()

    override this.Initialize() =
        this.Styles.Add(FluentTheme())
        this.Styles.Load "avares://EasyBuild.CommitLinter/Styles.xaml"

    // this.RequestedThemeVariant <- Styling.ThemeVariant.Default

    override this.OnFrameworkInitializationCompleted() =
        match this.ApplicationLifetime with
        | :? IClassicDesktopStyleApplicationLifetime as desktopLifetime ->
            desktopLifetime.MainWindow <- MainWindow config
        | _ -> ()
```

**Comment by JaggerJo** [comment]
Yeah, I can reproduce it.

Really seems to be an issue with all Subscriptions that are created from properties. We can fix this in FuncUI.
 
<img width="1033" alt="Screenshot 2024-09-05 at 16 40 28" src="https://github.com/user-attachments/assets/359c8cb9-6fa2-43be-ac5b-bd0eccf00e16">

work around for now:

```fsharp
            TextBox.onTextChanged (
                (fun text ->
                    if text <> shortDescription then
                        dispatch (ChangeShortDescription text)
                    else
                        ()
                ),
                SubPatchOptions.OnChangeOf shortDescription
            )
```

**Comment by JaggerJo** [comment]
@MangelMaxime this has been fixed in version 1.5.1

**Comment by MangelMaxime** [comment]
Awesome thank you
